### ISSUE TITLE
Zoom in photos without opening them

### ISSUE THREAD

**Issue by olekstomek** [tuskyapp/Tusky#3113]
In order to be able to enlarge a photo (zoom-in) I must first click on the photo and then I can enlarge it. It would also be convenient to add the ability to quickly zoom in on a photo - by stretching your fingers (this is how it works in the Instagram application). Thanks to this, I don't have to open the photo to enlarge it, but I can enlarge it right away.
(https://github.com/mastodon/mastodon-android/issues/479)

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by connyduck** [comment]
Yes that would be convienent but it also means loading all images fullsize immediately which would make Tusky use way more data.

**Comment by MKCOOL142** [comment]
Couldn't you load the full size one while the user zooms in rather than before they zoom in?

**Comment by connyduck** [comment]
I guess, but the full size will take a while to load and with zooming one usually expects immediate feedback, I doubt it will make for a good user experience.

**Comment by MKCOOL142** [comment]
> I guess, but the full size will take a while to load and with zooming one usually expects immediate feedback, I doubt it will make for a good user experience.

Maybe you could IF the image is fully loaded allow zooming in, if only the low res one loaded you could simply prevent zooming until it loaded

**Comment by Lakoja** [comment]
This would (technically) be rather complicated from a good or consistent UI perspective.

Because network lag and/or server lag is a topic.

I wonder how Instagram does it?

What happens for example if I want to enlarge it and the loading (then) takes some time? There couldn't be a snappy enlargement but some sort of waiting animation (?) would need to be displayed only then.

(So the current "tap first, wait, enlarge" is at least consistent and predictable for the user.)

**Comment by MKCOOL142** [comment]
> This would (technically) be rather complicated from a good or consistent UI perspective.
> 
> Because network lag and/or server lag is a topic.
> 
> I wonder how Instagram does it?
> 
> What happens for example if I want to enlarge it and the loading (then) takes some time? There couldn't be a snappy enlargement but some sort of waiting animation (?) would need to be displayed only then.
> 
> (So the current "tap first, wait, enlarge" is at least consistent and predictable for the user.)

From what I've seen Telegram doesn't have a low and high res version so the user waits until the image fully loaded and only then is able to zoom in, before that is only a blurry image preview with a loading spinner

**Comment by olekstomek** [comment]
I think that on Instagram it is zoom without increasing of image quality. I checked that when I zoom-in picture my network doesn't send more data. And you can zoom-in [also video](https://help.instagram.com/1827844047447474) so it looks that is "technical zoom" like in webrowswer for example using ctr and + or ctrl and scroll on your mouse.
